### PR TITLE
[EPMEDU-867]: Show Tbilisi center on disabled gps location

### PR DESCRIPTION
**Pull request by Makentoshe**
Issue link: https://jira.epam.com/jira/browse/EPMEDU-867

**Tech changes:**
- Hardcode Tbilisi city center coordinates as the initial value
- Move LocationProvider and GpsSettingsProvider to geolocation module

### CODE REVIEW

File: feature/home/src/main/java/com/epmedu/animeal/home/presentation/viewmodel/HomeState.kt [hunk 1]
```diff
@@ -5,7 +5,7 @@ import com.epmedu.animeal.home.data.model.FeedingPoint
 import com.mapbox.maps.Style
 
 data class HomeState(
-    val currentLocation: MapLocation = MapLocation(),
+    val currentLocation: MapLocation = MapLocation.Initial,
```
Comment (Fileek) [inline feature/home/src/main/java/com/epmedu/animeal/home/presentation/viewmodel/HomeState.kt]: Why didn't you just change Initial to Tbilisi here?

Comment (Makentoshe) [reply feature/home/src/main/java/com/epmedu/animeal/home/presentation/viewmodel/HomeState.kt]: Ok, changed

File: app/src/main/java/com/epmedu/animeal/MainActivity.kt [hunk 2]
```diff
@@ -23,6 +23,11 @@ class MainActivity : ComponentActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
+        Thread.UncaughtExceptionHandler { t, e ->
```
Comment (egorikftp) [inline app/src/main/java/com/epmedu/animeal/MainActivity.kt]: Why we need this?

File: feature/home/src/main/java/com/epmedu/animeal/home/presentation/ui/MapboxMap.kt [hunk 3]
```diff
@@ -24,8 +25,8 @@ import com.mapbox.maps.plugin.scalebar.scalebar
 fun MapboxMap(state: HomeState) {
     val mapBoxView = mapView(state.mapBoxPublicKey, state.mapBoxStyleUri)
 
-    var initialLocationReceived by rememberSaveable { mutableStateOf(false) }
-    LaunchedEffect(Unit) { initialLocationReceived = false }
+    var initialUserLocationReceived by rememberSaveable { mutableStateOf(false) }
```
Comment (egorikftp) [inline feature/home/src/main/java/com/epmedu/animeal/home/presentation/ui/MapboxMap.kt]: Do we need this variables now?
I assume we can remove everything

File: library/geolocation/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -0,0 +1,2 @@
+<?xml version="1.0" encoding="utf-8"?>
+<manifest package="com.epmedu.animeal.geolocation" />
```
Comment (egorikftp) [inline library/geolocation/src/main/AndroidManifest.xml]: Please move package into gradle file using `namespace`

File: feature/home/src/main/java/com/epmedu/animeal/home/presentation/viewmodel/HomeViewModel.kt [hunk 5]
```diff
@@ -35,15 +37,21 @@ internal class HomeViewModel @Inject constructor(
...
-            gpsSettingsProvider.fetchUpdates().collect {
-                updateState { copy(gpsSettingState = it) }
-            }
+            gpsSettingsProvider.fetchUpdates().collect(::collectGpsSettings)
+        }
+    }
+
+    private fun collectGpsSettings(state: GpsSettingsProvider.GpsSettingState) {
+        val uiGpsState = when (state) {
+            GpsSettingsProvider.GpsSettingState.Enabled -> GpsSettingState.Enabled
```
Comment (egorikftp) [inline feature/home/src/main/java/com/epmedu/animeal/home/presentation/viewmodel/HomeViewModel.kt:51]: can be just `GpsSettingState.Enabled`

File: app/src/main/java/com/epmedu/animeal/MainActivity.kt (hunk 2 above)
Comment (Makentoshe) [reply app/src/main/java/com/epmedu/animeal/MainActivity.kt]: Sorry, it is only for debug on my device.

File: feature/home/src/main/java/com/epmedu/animeal/home/presentation/viewmodel/HomeViewModel.kt (hunk 5 above)
Comment (Makentoshe) [reply feature/home/src/main/java/com/epmedu/animeal/home/presentation/viewmodel/HomeViewModel.kt:51]: No-no-no, there are two different GpsSettingState. One from GpsSettingsProvider(model) and other just GpsSettingState(presenter). In other case we should write @Stable annotation. Here our ui and model are independent.

File: feature/home/src/main/java/com/epmedu/animeal/home/presentation/ui/MapboxMap.kt (hunk 3 above)
Comment (Makentoshe) [reply feature/home/src/main/java/com/epmedu/animeal/home/presentation/ui/MapboxMap.kt]: Ok, removed

File: app/src/main/java/com/epmedu/animeal/MainActivity.kt (hunk 2 above)
Comment (Makentoshe) [reply app/src/main/java/com/epmedu/animeal/MainActivity.kt]: Sorry, it is only for debug on my device.

**Review by egorikftp — APPROVED** [review]

**Pull request merged**